src/EnergyPlus/ZoneContaminantPredictorCorrector.cc

```diff
@@ -1665,27 +1665,41 @@ namespace ZoneContaminantPredictorCorrector {
                 ControlledCO2ZoneFlag = false;
                 // Check all the controlled zones to see if it matches the zone simulated
                 for (ContControlledZoneNum = 1; ContControlledZoneNum <= NumContControlledZones; ++ContControlledZoneNum) {
-                    if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
-                        ZoneAirCO2SetPoint = ZoneCO2SetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
-                        if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointOn) {
-                            ZoneAirCO2SetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointValue;
+                    if (ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum == ZoneNum) {
+                        if (GetCurrentScheduleValue(ContaminantControlledZone(ContControlledZoneNum).AvaiSchedPtr) > 0.0) {
+                            ZoneAirCO2SetPoint = ZoneCO2SetPoint(ContaminantControlledZone(ContControlledZoneNum).ActualZoneNum);
+                            if (ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointOn) {
+                                ZoneAirCO2SetPoint = ContaminantControlledZone(ContControlledZoneNum).EMSOverrideCO2SetPointValue;
+                            }
+                            ControlledCO2ZoneFlag = true;
+                            break;
```

Comment:
At first this `break` seemed odd to me.  I thought you'd want to keep looping to get all of them that apply, but that's not the case.  You only want to get the first one for a particular zone, but you do want to make sure all of the controllers (overall) are modeled.  This makes sense, but please confirm that is the intent.

Reply:
Yes.